lib/registries/addon/components/registries-license-picker/component.ts

```diff
@@ -0,0 +1,113 @@
+import { action } from '@ember-decorators/object';
+import { alias, sort } from '@ember-decorators/object/computed';
+import { service } from '@ember-decorators/service';
+import Component from '@ember/component';
+import { task, timeout } from 'ember-concurrency';
+import DS from 'ember-data';
+import I18N from 'ember-i18n/services/i18n';
+import Toast from 'ember-toastr/services/toast';
+
+import { layout } from 'ember-osf-web/decorators/component';
+import License from 'ember-osf-web/models/license';
+import { QueryHasManyResult } from 'ember-osf-web/models/osf-model';
+import Registration from 'ember-osf-web/models/registration';
+import Analytics from 'ember-osf-web/services/analytics';
+import defaultTo from 'ember-osf-web/utils/default-to';
+import styles from './styles';
+import template from './template';
+
+@layout(template, styles)
+export default class RegistriesLicensePicker extends Component.extend({
+    queryLicenses: task(function *(this: RegistriesLicensePicker, name?: string) {
+        if (!(this.providerLicenses && this.providerLicenses.length)) {
+            return;
+        }
+
+        if (name) {
+            yield timeout(500);
+        }
+
+        const licensesAcceptable = this.providerLicenses
+            .filter(license => license.get('name').includes(name || ''));
+
+        this.setProperties({ licensesAcceptable });
+        this.registration.notifyPropertyChange('license');
+
+        return licensesAcceptable;
+    }).on('didReceiveAttrs').restartable(),
```

Comment:
So the nice thing here is that, when the user clicks the select trigger they will see a list with all provider licenses and then they can type in a name or part of it to filter through the list. It can be done in a better way though.